signing-and-verification-workflow.md

```diff
@@ -0,0 +1,52 @@
+    1. Verify that the signing certificate is valid and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+    1. Verify that the signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+    1. Generate signature.
+        1. Generate signature using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes).
+        1. Obtain an [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant timestamp for the signature generated in the previous step. This step is optional but recommended.
+            1. Verify that the timestamp signing certificate satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+            1. Verify that the timestamp signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+        1. Embed timestamp to the signature envelope.
+1. **Push the signature envelope:** Push the signature envelope generated in the previous step to the repository.
+1. **Generate signature artifact manifest:** As decribed in [signature specification](./signature-specification.md#storage) create the Notary v2 signature artifact manifest for the signature envelope generated in step 1.
+1. **Push signature artifact manifest:** Push Notary v2 signature artifact manifest to the repository.
+
+The user can push the OCI artifact to the repository before, after, or in between the signature generation process.
+
+## Verification workflow
+
+The user wants to pull OCI artifacts only if they are signed by the trusted publisher and the signature is valid.
+
+### Prerequisites
+
+- User has a digest of OCI artifact they want to download. If the user wants to use a tag to identify the artifact then that tag must be resolved to digest.
+- User has configured [trust store and trust policy](./trust-store-trust-policy-specification.md) required for signature verification.
+
+### Steps
+
+1. **Get signature artifact descriptors:** Using the ORAS [Manifest Referrers API](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md) download  Notary v2 signature artifact descriptors. The ORAS referrers API only needs artifact digest and artifact type(optional) as input. The Notary v2 signature's artifact type is `application/vnd.cncf.notary.v2.signature`.
+1. **Get signature artifact manifests:** For each signature artifact descriptor, perform the following steps:
+    1. **Get signature artifact manifests:** Download the Notary v2 signature's artifact manifests for the given artifact descriptor.
+    1. **Filter signature artifact manifests:**
+        1. Based on the value of the `blobs->mediaType` attribute and supported signature envelope format, filter out signature manifests.
```

Comment:
I'm not quite following the step here.
Once the user has the manifest, they download the blob for the signature payload. Today, we only have one format, so I'm not sure we need to call out filtering as we haven't identified if we'll have multiple blobs in a single signature or multiple signatures for different formats.

Reply:
Added it for backward compatibility so that in the future if we have multiple signature formats we don't break existing clients.

Reply:
Can you elaborate in the in the workflow about why this is done.